### PR TITLE
feat: Added support s3 endpoints

### PR DESCRIPTION
**Pull request by amanpruthi**
This Pr has been tested for a upgrade scenario as well as a fresh setup scenario.

### CODE REVIEW

File: modules/endpoint/main.tf [hunk 1]
```diff
@@ -0,0 +1,21 @@
+resource "aws_vpc_endpoint" "endpoint" {
```
Comment (jsbroks) [inline modules/endpoint/main.tf:1]:
```suggestion
resource "aws_vpc_endpoint" "default" {
```

**Review by jsbroks — APPROVED** [review]

File: main.tf [hunk 2]
```diff
@@ -57,6 +57,15 @@ locals {
   network_database_subnet_group_name   = var.create_vpc ? module.networking.database_subnet_group_name : "${var.namespace}-database-subnet"
 }
 
+module "s3_endpoint" {
+  count                  = length(var.private_link_allowed_account_ids) > 0 ? 1 : 0
+  source                 = "./modules/endpoint"
+  service_name           = "com.amazonaws.${data.aws_region.current.name}.s3"
+  network_id             = local.network_id
+  private_route_table_id = [module.networking.private_route_table_ids[0],module.networking.private_route_table_ids[1]]
```
Comment (jsbroks) [inline main.tf:65]:
```suggestion
  private_route_table_id =module.networking.private_route_table_ids
```

Does this not work?

**Review by jsbroks — CHANGES_REQUESTED** [review]

File: main.tf (hunk 2 above)
Comment (amanpruthi) [reply main.tf:65]: Yes worked

**Review by jsbroks — APPROVED** [review]

**Comment by jsbroks** [comment]
This PR is included in version 4.14.0 :tada:

**Pull request merged**